web-app/src/app/(landing)/(auth)/signup/data.ts

```diff
@@ -2,30 +2,35 @@ import { z } from "zod";
 
 import { FormData, FormIntlTranslation } from "@/lib/form";
 
-const signUpFormSchema = (t: FormIntlTranslation<"Auth.Pages.SignUp.Form">) =>
+const signUpFormSchema = (
+  t: FormIntlTranslation<"Auth.Pages.SignUp.Form"> | undefined = undefined,
+) =>
   z
     .object({
       username: z
         .string()
         .min(2, {
-          message: t("Errors.Username.min"),
+          message: t?.("Errors.Username.min"),
         })
-        .max(50, { message: t("Errors.Username.max") }),
+        .max(50, { message: t?.("Errors.Username.max") })
+        .regex(/^\S*$/, {
+          message: t?.("Errors.Username.regex"),
+        }),
       email: z.string().email({
-        message: t("Errors.Email.invalid"),
+        message: t?.("Errors.Email.invalid"),
       }),
       password: z
         .string()
         .min(8, {
-          message: t("Errors.Password.min"),
+          message: t?.("Errors.Password.min"),
         })
         .regex(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)/, {
```

Comment:
lets have that somewhere global, maybe as a verification function?

Reply:
Sorry the regex

Reply:
63ff21ee4c846e52e5b8b3a966c6e055b13993d8